servers/physics_3d/godot_collision_solver_3d_sat.cpp

```diff
@@ -843,7 +843,7 @@ static void _collision_sphere_box(const GodotShape3D *p_a, const Transform3D &p_
 	// Find the point on the box nearest to the center of the sphere.
 
 	Vector3 center = p_transform_b.xform_inv(p_transform_a.origin);
-	Vector3 extents = box_B->get_half_extents();
+	Vector3 extents = box_B->get_half_extents() * p_transform_b.basis[0].length();
```

Comment:
It looks like the `p_transform_b.xform_inv()` works if the transform consists of only translation and rotation elements, if it includes scaling then the result is bogus.

To make it work I changed the two lines as follows:
```
	Vector3 center = p_transform_b.affine_inverse().xform(p_transform_a.origin);
	Vector3 extents = box_B->get_half_extents();
```

The affine_inverse() produces a inversion-transform which works for translation/rotation/scale. The result is that center is correctly translated to the box_B coordinate system.